### PR TITLE
Add @mockable support

### LINKED ISSUE #28: Implement @mockable support
ZIO guidelines suggest using zio-macros to generate helper methods for service mocking. 
It would be great to add hints in Intellij Idea for the generated code.

### PR DESCRIPTION
**Pull request by timaliberdov**
Closes #28

### CODE REVIEW

**Comment by timaliberdov** [comment]
ZStream mocks generation should be fixed in https://github.com/zio/zio/pull/3895, ZSink generation still doesn't work in zio's @mockable macro so I didn't implement it in the plugin

File: src/main/scala/zio/intellij/synthetic/macros/MockableInjector.scala [hunk 1]
```diff
@@ -0,0 +1,243 @@
...
+
+  private def sortOverloads(infos: Seq[MemberInfo]): Seq[MemberInfo] = {
+    import scala.math.Ordering.Implicits.seqDerivedOrdering
+    infos.sortBy(_.signature)(Ordering[Seq[String]].on {
+      case FunctionDeclaration(method) =>
+        for {
+          clause       <- method.paramClauses.clauses
+          param        <- clause.parameters
+          paramType    <- param.paramType
+          presentation = presentationString(paramType.typeElement)
```
Comment (hmemcpy) [inline src/main/scala/zio/intellij/synthetic/macros/MockableInjector.scala]: `presentationString` is removed in 2020.2. Any other way to do this?

Comment (timaliberdov) [reply src/main/scala/zio/intellij/synthetic/macros/MockableInjector.scala]: Hmm.. I'll look for an alternative. Note to self: same for accessible macro

Comment (hmemcpy) [reply src/main/scala/zio/intellij/synthetic/macros/MockableInjector.scala]: Yeah, I noticed it there too!

Thanks!

Comment (timaliberdov) [reply src/main/scala/zio/intellij/synthetic/macros/MockableInjector.scala]: Borrowed rendering related code from 2020.1 and 2020.2 versions of Scala plugin 😅
We'll need some refactoring once 2020.2 is out of beta.

**Pull request merged**